### PR TITLE
Announcements integration

### PR DESCRIPTION
**Pull request by chrisnojima**
This hooks up announcements in the people tab. There is a new admin panel to turn on announcements (there are 3 there already for admin only). You can play with that to test. Note only one will show up at a time so you should just deactivate any other ones. You also likely have to restart your daemon to get it to show up

There's a bug in the server so the id we get is 0 which causes dismissal to not work but that'll be fixed soon. this can be reviewed i think

- [x] test dismissal (needs server deploy)

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers

File: shared/people/announcement/container.js [hunk 1]
```diff
@@ -1,35 +1,83 @@
...
 const mapStateToProps = () => ({})
-// TODO
-const mapDispatchToProps = () => ({
-  onConfirm: () => {
-    console.log('announcement onConfirm TODO')
+
+// Really the saga should handle all of this and we shouldn't have any of these ownProps passed in but this
+// is how the other types work in this list. TODO change this to be more modern
+const mapDispatchToProps = dispatch => ({
+  _onConfirm: (id, appLink, url) => {
```
Comment (buoyad) [inline shared/people/announcement/container.js]: you can pull `ownProps` in here and avoid the closures in `mergeProps`

File: shared/constants/people.js [hunk 2]
```diff
@@ -159,16 +158,17 @@ export const reduceRPCItemToPeopleItem = (
...
       )
     }
-  } else if (item.data.t === 3) {
-    if (__DEV__ && flags.peopleAnnouncementsEnabled) {
-      const a: any = item.data.announcement
+  } else if (item.data.t === RPCTypes.homeHomeScreenItemType.announcement) {
+    const a = item.data.announcement
+    if (a) {
       return list.push(
         makeAnnouncement({
```
Comment (buoyad) [inline shared/constants/people.js]: looks like `makeAnnouncement` below is dropping some fields

Comment (chrisnojima) [reply shared/constants/people.js]: fixed

File: shared/constants/people.js [hunk 3]
```diff
@@ -184,6 +184,7 @@ export const makeAnnouncement: I.RecordFactory<Types._Announcement> = I.Record({
   badged: false,
   confirmLabel: null,
   dismissable: false,
```
Comment (buoyad) [inline shared/constants/people.js]: `iconUrl` here too

Comment (chrisnojima) [reply shared/constants/people.js]: k fixed ,txn

**Pull request merged**